manifests/init.pp

```diff
@@ -284,6 +284,9 @@
   Hash[String[1], Data] $agent6_extra_options = {},
   $agent5_repo_uri = $datadog_agent::params::agent5_default_repo,
   $agent6_repo_uri = $datadog_agent::params::agent6_default_repo,
+  $use_apt_backup_keyserver = $datadog_agent::params::use_apt_backup_keyserver,
+  $apt_backup_keyserver = 'hkp://pool.sks-keyservers.net:80',
+  $apt_keyserver = 'hkp://keyserver.ubuntu.com:80',
```

Comment:
`hkp://keyserver.ubuntu.com:80` should probably be in `$datadog_agent::params`.

Also, we should be adding new params using abstract data types:
`Optional[String]` and `Optional[Boolean]` or `Variant[Enum['true', 'false'], Boolean]` (see: https://puppet.com/docs/puppet/5.3/lang_data_abstract.html)

Reply:
Yes, `use_apt_backup_keyserver` should be optional. What is the difference between putting a variable in `init.pp` and in `params.pp` ? The keyserver url is not meant to be user facing and is not optional because we need one to fetch the pgp key